### ISSUE TITLE
Clarify naming convention for model files

### ISSUE THREAD

**Issue by mihai-sysbio** [MetabolicAtlas/standard-GEM#38]
All model files must be named based on the repository name and versioned explicitly. Example: yeast-GEM-0.1.0.json. It must also include a release.md file that describes the release notes of the different versions of the model and the data/code used to generate the new version from the previous one. This will help standardize the model generation/ customization process and thus improve the reproducibility and maintainability.

**Comment by mihai-sysbio** [comment]
Reflecting back on this, I think having the version encoded _within_ the file is superior to relying on the file name.
Since this is already in line with the SBML, YAML and MatLab standards and practices, I will proceed to close this issue.

**Comment by draeger** [comment]
What happened to the `i` prefix (for *in silico*) that has become a widely recognized convention for many genome-scale metabolic models? Unless there is a compelling reason to deviate from it, I would argue that extending established community conventions is preferable to introducing incompatible alternatives.

This publication proposed a modernized naming convention that has subsequently been adopted by many modelers:

https://link.springer.com/article/10.15252/msb.20199235

**Comment by mihai-sysbio** [comment]
Thank you @draeger for bringing this up, it's an interesting question. I think there has been progress across all the tables in the publication.

Looking at the ideas in [_Table 4_](https://link.springer.com/article/10.15252/msb.20199235/tables/4), specifically at:
> **Model**: Recognized naming convention o iPfal19: i + species indicator + iteration identifier 

there is large conceptual overlap:
- there needs to be a species indicator in the model name
- the version needs to be captured
- there need to be an "indication" of what this file is

The progress regarding the version is that it's arguably been made more robust by:
- recording the version inside the file so that it cannot simply be lost when renaming the file
- keeping a more constant name there is better persistence in links or workflows when models update

Regarding the prefix `i`, that has been more "relaxed"/expanded through the suffix `-GEM` or `-GSMM` that is more descriptive of the modelling paradigm. The other benefit of a suffix is that it makes for, in my view, more intuitive alphabetical sorting.

**Comment by draeger** [comment]
Thanks, that makes sense, especially regarding the versioning.

My original point was less about the technical aspects and more about preserving established community conventions where possible. I can certainly see the advantages of the -GEM suffix, though it does imply a transition away from a naming scheme that many modelers have become familiar with over the years.

**Comment by mihai-sysbio** [comment]
Do you have any ideas on how to address this? GitHub has a pretty decent mechanism to rename repos, by maintaining a redirect from the old name to the new name where possible.

**Comment by draeger** [comment]
My concern is actually broader than the _i_ prefix. While reviewing the manuscript, I realized that requiring repository names and model names to coincide may not be appropriate in general. Repository names should remain stable, whereas model names often evolve over time and a single repository may legitimately contain multiple related models (e.g., pan-GEM projects or strain-specific reconstructions). I’ll leave more detailed comments on this point directly in the manuscript.

**Comment by mihai-sysbio** [comment]
Oh I see, maybe there is then some conceptual overlap with #37

**Comment by mihai-sysbio** [comment]
I've been reflecting on this, and there is some depth to the underlying expectation that a repository name matches the model name.
If there is no such expectation, then there is little to prohibit many models be present in the `model` folder. At the same time, it becomes difficult to programatically predict what those models would be names.

A concrete example is https://github.com/iAMB-RWTH-Aachen/Ustilago_maydis-GEM/tree/d8f1c26735598d5662bb4ad3013880e406189ff1/model where the different names prohibits the validation to run. In addition to that, when presenting the model, it becomes hard to talk about it concretely - as the model name and repo name are different, one has to be very precise.

For pan-GEM projects, standard-GEM indeed does not specify a concrete way.

I would be tempted to expect a index-style overview that would not only contain the models but also the relationship between them, e.g., ranging from non-interacting collection to an organism of interacting tissue models.

**Comment by draeger** [comment]
Yes, I think an index-style overview would actually be a very good solution. It would make the relationship between repository and model(s) explicit and machine-readable rather than relying on naming conventions.

I would prefer this over requiring repository and model names to match, particularly since established model naming conventions often include iteration/version information that should not become part of the stable repository name. Even requiring a common prefix may sometimes be problematic, e.g., when model names include author initials that may not yet be known when the repository is created.

Such an index would also provide a natural solution for pan-GEMs and other repositories containing multiple related models.

**Comment by mihai-sysbio** [comment]
I would very much like to work on this index, but it feels like it's a "to go far, go together" moment. In other words, I would rather engage the wider community for have consensus on how such a file be structured, also in a way that is compatible with all the different file formats. Therefore, it would feel most natural for me to have this as the basis for a 2.0 version of standard-GEM.

**Comment by draeger** [comment]
That sounds like a very sensible approach to me. I agree that the structure of such an index should be discussed with the wider community rather than defined ad hoc, and making this a major feature of standard-GEM 2.0 seems appropriate.

For the current version, I would therefore simply avoid making an exact match between repository and model names a strict requirement, leaving room for this more general solution in 2.0.
